core/src/main/scala/kafka/raft/RaftManager.scala

```diff
@@ -318,4 +329,28 @@ class KafkaRaftManager[T](
   override def voterNode(id: Int, listener: String): Option[Node] = {
     client.voterNode(id, listener).toScala
   }
+
+  def verifyCommonFields(
+    providedClusterId: String,
+  ): Unit = {
+    if (!providedClusterId.equals(clusterId)) {
+      throw new InconsistentClusterIdException("Provided cluster ID " + providedClusterId +
```

Comment:
This applies to all of the `throw new` added.

If we throw here, this will get caught here and print an error message, right?
https://github.com/apache/kafka/pull/16058/files#diff-91060c918c99d25342f625c146f14425716eda9d8fcfe1126b2c45feff388362R152-R154

How about for now we just complete the requests with the unsupported version error responses?

Also, I have to double check but I think cluster ids are check by the `KafkaRaftClient` and not the `KafkaRaftManager`. We have this abstraction so we can test that behavior in the `KafkaRaftClientTest` suite or if we add a new test suite for this functionality.

Most our interesting tests are against `KafkaRaftClient`. The tests for `KafkaRaftManager` are more basic and limited.

Reply:
> If we throw here, this will get caught here and print an error message, right?
https://github.com/apache/kafka/pull/16058/files#diff-91060c918c99d25342f625c146f14425716eda9d8fcfe1126b2c45feff388362R152-R154

Hmm, your URL doesn't work for me. As a general matter, exceptions not otherwise handled in ControllerApis.scala will just invoke `AbstractRequest.getErrorResponse`.

> How about for now we just complete the requests with the unsupported version error responses?

OK

Reply:
> Hmm, your URL doesn't work for me.

This is the URL/code I was trying to send: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/ControllerApis.scala#L147-L151